maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/LiferayMavenModuleProjectProvider.java

```diff
@@ -0,0 +1,493 @@
+                    final IProjectConfigurationManager projectConfigurationManager = MavenPlugin.getProjectConfigurationManager();
+
+                    projectConfigurationManager.importProjects( projectInfos, importConfiguration, monitor );
+
+                    IProject project = workspace.getRoot().getProject( projectName );
+
+                    if ( project.exists() )
+                    {
+                        newProjects.add( project );
+
+                        if( !CoreUtil.isNullOrEmpty( newProjects ) )
+                        {
+                            for( IProject iProject : newProjects )
+                            {
+                                projectNames.insert().setName( iProject.getName() );
+                            }
+                        }
+
+                        if( CoreUtil.isNullOrEmpty( newProjects ) )
+                        {
+                            retval = LiferayMavenCore.createErrorStatus( "New project was not created due to unknown error" );
+                        }
+                        else
+                        {
+                            for( final IProject iProject : newProjects )
+                            {
+                                try
+                                {
+                                    projectConfigurationManager.updateProjectConfiguration( new MavenUpdateRequest(
+                                        iProject, mavenConfiguration.isOffline(), true ), monitor );
+                                }
+                                catch( Exception e )
+                                {
+                                    LiferayMavenCore.logError( "Unable to update configuration for " + project.getName(), e );
+                                }
+                            }
+                        }
+
+                    }
+                    else
+                    {
+                        retval = LiferayMavenCore.createErrorStatus( "Unable to import this Project " );
+                    }
+
+                }
+            }
+            catch( IOException e1 )
+            {
+                retval = LiferayMavenCore.createErrorStatus( "Unable to create this Project " );
+            }
+        }
+
+        if( retval == null )
+        {
+            retval = Status.OK_STATUS;
+        }
+
+        return retval;
+    }
+
+    private void setBasedirRename( MavenProjectInfo projectInfo ) throws IOException
+    {
+        File workspaceRoot = ResourcesPlugin.getWorkspace().getRoot().getLocation().toFile();
+        File basedir = projectInfo.getPomFile().getParentFile().getCanonicalFile();
+
+        projectInfo.setBasedirRename( basedir.getParentFile().equals( workspaceRoot )
+            ? MavenProjectInfo.RENAME_REQUIRED : MavenProjectInfo.RENAME_NO );
+    }
+
+    @Override
+    public <T> List<T> getData( String key, Class<T> type, Object... params )
```

Comment:
Isn't this an exact copy of the getData() from the other maven project provider?  If so, this amount of code copying is not good in general and should be avoided through class reuse.